xapian-core/backends/glass/glass_version.cc

```diff
@@ -103,7 +103,7 @@ GlassVersion::read()
 	if (rare(lseek(fd, offset, SEEK_SET) < 0)) {
 	    string msg = "Failed to rewind file descriptor ";
 	    msg += str(fd);
-	    throw Xapian::DatabaseOpeningError(msg, errno);
+	    throw Xapian::DatabaseNotFoundError(msg, errno);
```

Comment:
I think this case isn't really "not found" - it's reading the version file for a single file database, and the fd was either passed in or opened by `backends/dbfactory.cc`so we know the file exists.  It might not be a valid Xapian database, but that I think either `DatabaseOpeningError` or `DatabaseCorruptError` would be more appropriate for that case.

It's actually hard to trigger this case in normal use since the constructor already checked that fd is seekable.  Probably you'd have to close the file descriptor behind Xapian's back, and then this would fail with `EBADF` ("Bad file descriptor"), but other things might fail first as the same file descriptor is used to read the tables.

Reply:
Thanks for detailed explanation. I will push the changes.